### ISSUE TITLE
Behaviour of c=None is inconsistent in 2.0

### ISSUE THREAD

**Issue by Zaharid** [matplotlib/matplotlib#7899]
The following code
```python
import numpy as np
import matplotlib.pyplot as plt
f,ax = plt.subplots()
x = np.arange(10)
y = 2*x

plt.errorbar(x,y, c=None)
plt.errorbar(x,2*y, c=None)
```

Produces two lines with the same color (first one of the cycle). In 1.5 it is two different colors.
It is also two colors if I change `errorbar` to `plot`.

**Comment by dstansby** [comment]
This bisects back to c8a365a383880d4b4c87208139490297cead741c

**Comment by tacaswell** [comment]
Sigh, the problem is that there is a user supplied color, so the color cycle is being ignored in errorbar and falling through to the default line color.

**Comment by tacaswell** [comment]
@Zaharid Sorry for my whining above.

Thank you for a clear bug report!  Very much appreciate it :smile: .

**Comment by jenshnielsen** [comment]
Should this be closed now that #8037 is merged?

**Comment by dstansby** [comment]
Yep, looks fixed to me.
